vendor/engines/saas/lib/saas/before_action.rb

```diff
@@ -25,6 +25,14 @@ def check_account
           end
 
           failure :unauthorized do
+            if params[:subscription_canceled]
+              throw :warden, action: 'unauthenticated_canceled'
+            end
+
+            if params[:trial_expired] && !params[:subscription_canceled]
```

Comment:
`&& !params[:subscription_canceled]` here should be unnecessary since we already threw for that case.